extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -193,6 +207,80 @@ export class AmpRecaptchaService {
     }
   }
 
+  /**
+   * Function to create our bootstrap iframe.
+   *
+   * @param {string} sitekey
+   * @return {!Promise<!Element>}
+   * @private
+   */
+  createRecaptchaFrame_(sitekey) {
+
+    const iframe = this.win_.document.createElement('iframe');
+
+    return this.getRecaptchaFrameSrc_().then(recaptchaFrameSrc => {
+      iframe.src = recaptchaFrameSrc;
+      iframe.ampLocation = parseUrlDeprecated(this.win_.location.href);
+      iframe.setAttribute('scrolling', 'no');
+      iframe.setAttribute('data-amp-3p-sentinel', 'amp-recaptcha');
+      iframe.setAttribute('name', JSON.stringify(dict({
+        'sitekey': sitekey,
+        'sentinel': 'amp-recaptcha',
+      })));
+      setStyle(iframe, 'border', 'none');
+      /** @this {!Element} */
+      iframe.onload = function() {
+        // Chrome does not reflect the iframe readystate.
+        this.readyState = 'complete';
+      };
+
+      return iframe;
+    });
+  }
+
+  /**
+   * Function to get our recaptcha iframe src
+   *
+   * This should take the current URL,
+   * either in canonical (www.example.com),
+   * or in cache (www-example-com.cdn.ampproject.org),
+   * and get the curls subdomain (www-example-com)
+   * To then create the iframe src:
+   * https://www-example-com.recaptcha.my.cdn/rtv/recaptcha.html
+   *
+   * @return {!Promise<string>}
+   * @private
+   */
+  getRecaptchaFrameSrc_() {
+    if (getMode().localDev || getMode().test) {
+      return tryResolve(() => {
+        return getDevelopmentBootstrapBaseUrl(this.win_, 'recaptcha');
+      });
+    }
+
+    // Need to have the curls subdomain match the original document url.
+    // This is verified by the recaptcha frame to
+    // verify the origin on its messages
+    let curlsSubdomainPromise = undefined;
+    if (isProxyOrigin(this.win_.location.href)) {
+      curlsSubdomainPromise = tryResolve(() => {
+        return this.win_.location.hostname.split('.')[0];
+      });
+    } else {
+      curlsSubdomainPromise =
+        ampToolboxCacheUrl.createCurlsSubdomain(this.win_.location.href);
+    }
+
+    return curlsSubdomainPromise.then(curlsSubdomain => {
+      const recaptchaFrameSrc = 'https://' + curlsSubdomain +
+        `.recaptcha.${urls.thirdPartyFrameHost}/$internalRuntimeVersion$/` +
```

Comment:
I believe you and @lannka ran this test before. But want to make sure that the `recaptcha.html` is served and the the cursSubdomain value doesn't matter. Does the `$internalRuntimversion$` matters?

Reply:
@zhouyx We did test this. I tested this on Google's Cache, and @lannka Tested this on cloudflare.

We noticed that the `curlsSubdomain`, `$internalRuntimversion$` doesn't make a difference, and will serve the file nonetheless. 

Concerning the `curlsSubdomain`, we tested this on the domain: `*.ampproject.net/$internalRuntimeVersion$/recaptcha.html`. We made a change to get the `*.recpatcha.ampproject` to work as well, using the same wildcard rules, thus should definitely work. Not entirely sure when the domain change will land.

However, concerning `$internalRuntimversion$`, it should make a difference. In the case where it does not match, I am not sure which version it is serving from. But specifying the version now can save a ton of headaches down the line.